TypeCobol/Compiler/Scanner/ScannerStep.cs

```diff
@@ -202,15 +204,53 @@ public static class ScannerStep
             return tokensLinesChanges;
         }
 
+        /// <summary>
+        /// Check if two Tokens set are different
+        /// </summary>
+        /// <param name="curTokens">Current Tokens set</param>
+        /// <param name="newTokens">New Token Set</param>
+        /// <returns>true if different, false otherwise</returns>
+        private static bool AreTokensChanging(Token[] curTokens, Token[] newTokens)
+        {
+            if (curTokens == newTokens)
+                return false;
+            if (curTokens.Length != newTokens.Length)
+                return true;
+            for (int i = 0; i < curTokens.Length; i++)
+                if (!object.Equals(curTokens[i], newTokens[i]))
+                    return true;
+            return false;
+        }
         private static void ScanTokensLineWithMultilineScanState(int lineToScanIndex, TokensLine lineToScan, TextSourceInfo textSourceInfo, ISearchableReadOnlyList<TokensLine> documentLines, PrepareDocumentLineForUpdate prepareDocumentLineForUpdate, TypeCobolOptions compilerOptions, List<RemarksDirective.TextNameVariation> copyTextNameVariations, IList<DocumentChange<ITokensLine>> tokensLinesChanges, MultilineScanState initialScanState, out int nextLineToScanIndex, out TokensLine nextLineToScan)
         {
             // Scan the current line (or continuation lines group)
             MultilineScanState scanState = ScanTokensLineWithContinuations(lineToScanIndex, lineToScan, textSourceInfo, documentLines, prepareDocumentLineForUpdate, compilerOptions, copyTextNameVariations, tokensLinesChanges, initialScanState, out nextLineToScanIndex, out nextLineToScan);
 
             // Scan the following lines until we find that the scan state at the beginning of the next line has been updated
-            while (nextLineToScan != null && nextLineToScan.InitialScanState != null && !nextLineToScan.InitialScanState.Equals(scanState))
+            while (nextLineToScan != null && nextLineToScan.InitialScanState != null)
             {
-                scanState = ScanTokensLineWithContinuations(nextLineToScanIndex, nextLineToScan, textSourceInfo, documentLines, prepareDocumentLineForUpdate, compilerOptions, copyTextNameVariations, tokensLinesChanges, scanState, out nextLineToScanIndex, out nextLineToScan);
+                if (nextLineToScan.InitialScanState.Equals(scanState))
+                {//JCM: #1229 : We should break ==> but we must before check that tokens are not changing.
+                    TokensLine currentScanLine = nextLineToScan;
```

Comment:
Same question:
Is it specific to multiline comment?

Reply:
Ensuring that after an incremental change that the multilinescanstate is not changing is not enough, because a change can impact lines that were already in the desired multilinescansate but there tokens have been changed, so its true this situation appears often with multiline comments  that can hide syntactical source code, but not only.

Reply:
Ok.
For the new issue about incremental bug, can you also update documentation:
https://github.com/TypeCobolTeam/TypeCobol/wiki/TypeCobolIncrementalPipeline